funsite/templates/lms/index.html

```diff
@@ -1,257 +1,278 @@
 ## mako
+<%inherit file="funsite/parts/base-fixed-width.html" />
+<%namespace name='static' file='/static_content.html'/>
+<%namespace name="newsfeed" file="/newsfeed/article/block.html"/>
 <%!
 from django.core.urlresolvers import reverse
 from django.utils.formats import date_format
 from django.utils.translation import ugettext as _
 
 from courses.models import CourseSubject, Course
+from django.core.cache import cache
 from newsfeed.views import top_news
 from universities.models import University
-%>
-<%namespace name='static' file='/static_content.html'/>
-<%inherit file="funsite/parts/base-fixed-width.html" />
 
-<%block name="title">${_(u"The Freedom to Study")}</%block>
+CACHE_KEY = 'fun-apps-funsite-index-content'
+CACHE_EXPIRY = 60*60*24*30
+def django_cache(fn):
+    """A decorator to cache a def for everybody except staff users."""
+    def get_or_set(context, *args, **kw):
+        if context['request'].user.is_staff:
+            content = runtime.capture(context, fn, *args, **kw)
+            # Since we calculated a fresh page, we might as well update the version in cache
+            cache.set(CACHE_KEY, content, CACHE_EXPIRY)
+            return content
+        else:
+            content = cache.get(CACHE_KEY)
+            if not content:
+                content = runtime.capture(context, fn, *args, **kw)
+                cache.set(CACHE_KEY, content, CACHE_EXPIRY)
+            return content
+    return get_or_set
+%>
 
-<%block name="js_extra">
```

Comment:
The diff below is awsul but I'm "just" wrapping the content in a `def` (without any modification) to be able to cache it.